jmxtrans-core/src/main/java/com/googlecode/jmxtrans/connections/JMXConnection.java

```diff
@@ -32,20 +34,45 @@
 import javax.management.remote.JMXConnector;
 import java.io.Closeable;
 import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 @ToString
 @ThreadSafe
 public class JMXConnection implements Closeable {
 	@Nullable private final JMXConnector connector;
 	@Nonnull @Getter private final MBeanServerConnection mBeanServerConnection;
+	@Nonnull private final ExecutorService executor;
+	private boolean markedAsDestroyed;
+	private static final Logger logger = LoggerFactory.getLogger(JMXConnection.class);
 
 	public JMXConnection(@Nullable JMXConnector connector, @Nonnull MBeanServerConnection mBeanServerConnection) {
 		this.connector = connector;
 		this.mBeanServerConnection = mBeanServerConnection;
+		this.markedAsDestroyed = false;
+
+		executor = Executors.newSingleThreadExecutor();
+	}
+
+	public boolean isAlive(){
+		return !markedAsDestroyed;
 	}
 
 	@Override
 	public void close() throws IOException {
-		if (connector != null) connector.close();
+		markedAsDestroyed = true;
+		if (connector != null) {
+			executor.submit(new Runnable() {
```

Comment:
Why delegate the closing to an executor?

Reply:
I wrote it in commit message: when you call close it tries to notify server about it, so if your server is down you will wait until connection timed out.
This is just for fast release object from pool => fast release thread from queryExecutor thread pool.

Reply:
Ok, that make sense. Adding those thoughts as an inline comment would be nice.

Reply:
I've added comment. Check it out.